tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/ApalacheOper.scala

```diff
@@ -102,10 +102,15 @@ object ApalacheOper {
   }
 
   /**
-   * Attempt to dynamically cast an Int -> T function to a Seq(T). The first argument should be the function expression
-   * and the second argument should be an integer, denoting the maximal length of the sequence.
+   * <p>Attempt to dynamically cast an Int -> T function to a Seq(T). The first argument should be the function
+   * expression and the second argument should be an integer, denoting the maximal length of the sequence.</p>
+   *
+   * <p>We keep this operator in the IR. However, we are using the definition of this operator from Apalache.tla. Hence,
+   * if you construct an expression that contains `funAsSeq`, its constructor will throw.</p>
    */
   object funAsSeq extends ApalacheOper {
+    require(false, "This operator is defined in Apalache.tla. Do not construct it.")
```

Comment:
Why not just delete the object, if you're going to add `require(false)`?

Reply:
I also wanted to delete the object. But it is better to keep it as documentation.